src/mono/browser/runtime/driver.c

```diff
@@ -441,9 +441,7 @@ mono_wasm_init_finalizer_thread (void)
 {
 	// in the single threaded build, finalizers periodically run on the main thread instead.
 #ifndef DISABLE_THREADS
-	MONO_ENTER_GC_UNSAFE;
```

Comment:
Why it's safe now?

Reply:
it's duplicated logic

![image](https://github.com/dotnet/runtime/assets/271576/dccadaea-5a28-4bc2-810d-6eed67a93bda)